arangod/Replication2/ReplicatedLog/AgencyLogSpecification.cpp

```diff
@@ -0,0 +1,237 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2021-2021 ArangoDB GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Lars Maier
+////////////////////////////////////////////////////////////////////////////////
+
+#include "AgencyLogSpecification.h"
+
+#include "Basics/Exceptions.h"
+#include "Basics/StaticStrings.h"
+#include "Basics/application-exit.h"
+#include "Logger/LogMacros.h"
+#include "Logger/Logger.h"
+
+#include <velocypack/Iterator.h>
+#include <velocypack/velocypack-aliases.h>
+
+#include <type_traits>
+
+using namespace arangodb;
+using namespace arangodb::replication2;
+using namespace arangodb::replication2::agency;
+
+template<>
+struct velocypack::Extractor<LogTerm> {
+  static auto extract(VPackSlice slice) -> LogTerm {
+    return LogTerm{slice.getNumericValue<std::size_t>()};
+  }
+};
+
+template<>
+struct velocypack::Extractor<LogIndex> {
+  static auto extract(VPackSlice slice) -> LogIndex {
+    return LogIndex{slice.getNumericValue<std::size_t>()};
+  }
+};
+
+template<>
+struct velocypack::Extractor<LogId> {
+  static auto extract(VPackSlice slice) -> LogId {
+    return LogId{slice.getNumericValue<std::size_t>()};
+  }
+};
+
+template<>
+struct velocypack::Extractor<RebootId> {
```

Comment:
The `RebootId` type is defined in Cluster/ClusterTypes.h.
Should we move the Extractor definition there?

Reply:
Done